auth.ts

```diff
@@ -1,127 +1,144 @@
-        throw new Error('Error: Expected "event.headers.authorization" parameter to be set');
+      throw new Error(
+        'Expected "event.headers.authorization" parameter to be set',
+      )
     }
 
-    const match = tokenString.match(/^Bearer (.*)$/);
+    const match = tokenString.match(/^Bearer (.*)$/)
     if (!match || match.length < 2) {
-        throw new Error(`Error: Invalid Authorization token - ${tokenString} does not match "Bearer .*"`);
+      throw new Error(
+        `Invalid Authorization token - ${tokenString} does not match "Bearer .*"`,
+      )
     }
 
-    return match[1];
+    return match[1]
   }
 
-  private async verifyToken(event: APIGatewayProxyEvent): Promise<IScopeAndId> {
-    const token = this.extractBearerToken(event);
+  const verifyToken = async (
+    event: APIGatewayProxyEvent,
+  ): Promise<IScopeAndId> => {
+    const token = extractBearerToken(event)
 
-    const decoded: IDecodedToken = jwt.decode(token, { complete: true }) as IDecodedToken;
+    const decoded: IDecodedToken = jwt.decode(token, {
+      complete: true,
+    }) as IDecodedToken
     if (!decoded || !decoded.header || !decoded.header.kid) {
-      throw new Error('Error: Invalid Token');
+      throw new Error('Invalid Token')
     }
 
-    const rsaOrCertSigningKey: string = await this.getSigningKey(decoded.header.kid);
+    const rsaOrCertSigningKey: string = await getSigningKey(decoded.header.kid)
 
     const jwtOptions = {
-      audience: process.env.AUDIENCE,
-      issuer: process.env.TOKEN_ISSUER
+      audience,
+      issuer,
     }
-    const verifiedToken: IVerifiedToken = await jwt.verify(token, rsaOrCertSigningKey, jwtOptions) as IVerifiedToken;
+    const verifiedToken: IVerifiedToken = (await jwt.verify(
+      token,
+      rsaOrCertSigningKey,
+      jwtOptions,
+    )) as IVerifiedToken
 
-    const scopes: Array<string> = verifiedToken.scope.split(' ');
+    const scopes: Array<string> = verifiedToken.scope.split(' ')
 
     return {
       principleId: verifiedToken.sub,
-      scopes: scopes
+      scopes: scopes,
     }
   }
 
-  public async checkScopesAndResolve(event: APIGatewayProxyEvent, expectedScopes: [string]): Promise<string> {
-    try {
-      const verifiedToken = await this.verifyToken(event);
+  return async (
+    event: APIGatewayProxyEvent,
+    expectedScopes: [string],
+  ): Promise<string> => {
+    const verifiedToken = await verifyToken(event)
 
-      const scopes = verifiedToken.scopes;
+    const scopes = verifiedToken.scopes
 
-      if (!scopes) {
-        throw new Error('Error: No scopes supplied!');
-      }
+    if (!scopes) {
+      throw new Error('No scopes supplied!')
+    }
 
-      const scopesMatch = expectedScopes.some(scope => scopes.indexOf(scope) !== -1);
-      if(scopesMatch) {
-        return verifiedToken.principleId;
-      } else {
-        throw new Error('Error: You are not authorized!');
-      }
-    } catch (error) {
-      return error;
```

Comment:
Don't catch and return an error. A promise never resolves with an `Error` it has to reject with the `Error`. When you use `async/await` then just throw it. I just removed the catch block so the error will be thrown out of this function to be handled elsewhere.